internal/charmstore/store.go

```diff
@@ -1068,9 +1059,10 @@ func (s *Store) SetPromulgated(url *router.ResolvedURL, promulgate bool) error {
 
 // SetPerms sets the ACL specified by which for the base entity with the
 // given id. The which parameter is in the form "[channel].operation",
-// where channel, if specified, is one of "edge" or "stable" and operation
-// is one of "read" or "write". If which does not specify a channel then
-// the unpublished ACL is updated. This is only provided for testing.
+// where channel, if specified, is the string corresponding to one of the
+// ValidChannels and operation is one of "read" or "write". If which does
+// not specify a channel then the unpublished ACL is updated.
```

Comment:
It looks to me as if which must always specify a channel, otherwise the db update will be wrong.